index.js

```diff
@@ -80,71 +83,119 @@ module.exports = {
     ],
     "no-loss-of-precision": "error",
     "no-misleading-character-class": "error",
+    "no-new-native-nonconstructor": "error",
+    "no-new-symbol": "error",
     "no-obj-calls": "error",
     "no-promise-executor-return": "error",
     "no-prototype-builtins": "error",
-
-    // Disabled because when matching an exact number of spaces, writing them
-    // out is usually better than using the `{n}` syntax. It makes the regexp
-    // more similar to matched strings.
-    "no-regex-spaces": "off",
+    "no-self-assign": "error",
+    "no-self-compare": "error",
     "no-setter-return": "error",
     "no-sparse-arrays": "error",
-    "no-template-curly-in-string": "off",
+    "no-template-curly-in-string": "error",
+    "no-this-before-super": "error",
+    "no-undef": "error",
     "no-unexpected-multiline": "error",
+    "no-unmodified-loop-condition": "error",
     "no-unreachable": "error",
-    "no-unreachable-loop": "off",
+    "no-unreachable-loop": "error",
     "no-unsafe-finally": "error",
     "no-unsafe-negation": "error",
     "no-unsafe-optional-chaining": "error",
+    "no-unused-private-class-members": "off", // Not using private yet
+    "no-unused-vars": "error",
+
+    // This is going to cause problems, but I don't care.
+    "no-use-before-define": "error",
     "no-useless-backreference": "error",
+
+    // This fires too often.
     "require-atomic-updates": "off",
     "use-isnan": "error",
     "valid-typeof": "error",
 
-    // ----- Best Practices -----
+    // ----- Suggestions -----
 
+    // Read-only and write-only are a thing.
     "accessor-pairs": "off",
-
-    // Disabled because this is just guessing.
-    "array-callback-return": "off",
+    "arrow-body-style": "error",
 
     // Enabled, however this isn't strictly necessary because `var` declarations
     // are not allowed at all by `no-var`. But it can't hurt.
     //
     // @es5
     "block-scoped-var": "error",
-    // Disabled because class methods without `this` have legitimate uses, for
-    // example when a class has to implement a dummy empty method to conform to
-    // an interface.
-    "class-methods-use-this": "off",
+
+    // We have underscores in our public API already
+    "camelcase": "off",
+    "capitalized-comments": ["error", "always", {
+      ignorePattern: "c8",
+      ignoreConsecutiveComments: true,
+    }],
+    "class-methods-use-this": "error",
```

Comment:
Causes a few things that will have to be ignored (abstract methods) and a few places that we might want to move to static.